From c8c8c8775409a7ebff5b001fab47f10267a49e62 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Fri, 14 Feb 2025 12:29:36 +0100
Subject: [PATCH] Port to ISO C23
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Building with GCC 15, which moved to ISO C23, failed:

    async_io.c: In function ‘aio_walk_slots’:
    async_io.c:233:8: error: too many arguments to function ‘fn’; expected 0, have 1
      233 |       (*fn)(slot);
	  |       ~^~~~ ~~~~

and also warned about:

    d3des.c: In function ‘deskey’:
    d3des.c:58:6: warning: old-style function definition [-Wold-style-definition]
       58 | void deskey(key, edf)   /* Thanks to James Gillogly & Phil Karn! */
	  |      ^~~~~~

This patch fixes it.

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 async_io.c |   2 +-
 async_io.h |   2 +-
 d3des.c    |  27 ++++--------
 region.c   | 126 ++++++++++++++++++++++-------------------------------
 4 files changed, 63 insertions(+), 94 deletions(-)

diff --git a/async_io.c b/async_io.c
index 7723d08..2c12682 100644
--- a/async_io.c
+++ b/async_io.c
@@ -221,7 +221,7 @@ int aio_listen(int port, AIO_FUNCPTR initfunc, AIO_FUNCPTR acceptfunc,
  * Returns number of matching slots.
  */
 
-int aio_walk_slots(AIO_FUNCPTR fn, int type)
+int aio_walk_slots(void (*fn)(AIO_SLOT *slot), int type)
 {
   AIO_SLOT *slot, *next_slot;
   int count = 0;
diff --git a/async_io.h b/async_io.h
index 5524814..6404bb0 100644
--- a/async_io.h
+++ b/async_io.h
@@ -89,7 +89,7 @@ int aio_set_bind_address(char *bind_ip);
 int aio_add_slot(int fd, char *name, AIO_FUNCPTR initfunc, size_t slot_size);
 int aio_listen(int port, AIO_FUNCPTR initfunc, AIO_FUNCPTR acceptfunc,
                size_t slot_size);
-int aio_walk_slots(AIO_FUNCPTR fn, int type);
+int aio_walk_slots(void (*fn)(AIO_SLOT *slot), int type);
 void aio_call_func(AIO_FUNCPTR fn, int fn_type);
 void aio_close(int fatal_f);
 void aio_close_other(AIO_SLOT *slot, int fatal_f);
diff --git a/d3des.c b/d3des.c
index 8da059d..e8d10ab 100644
--- a/d3des.c
+++ b/d3des.c
@@ -55,9 +55,7 @@ static unsigned char pc2[48] = {
 	40, 51, 30, 36, 46, 54, 29, 39, 50, 44, 32, 47,
 	43, 48, 38, 55, 33, 52, 45, 41, 49, 35, 28, 31 };
 
-void deskey(key, edf)	/* Thanks to James Gillogly & Phil Karn! */
-unsigned char *key;
-int edf;
+void deskey(unsigned char *key, int edf)	/* Thanks to James Gillogly & Phil Karn! */
 {
 	register int i, j, l, m, n;
 	unsigned char pc1m[56], pcr[56];
@@ -92,8 +90,7 @@ int edf;
 	return;
 	}
 
-static void cookey(raw1)
-register unsigned long *raw1;
+static void cookey(unsigned long *raw1)
 {
 	register unsigned long *cook, *raw0;
 	unsigned long dough[32];
@@ -115,8 +112,7 @@ register unsigned long *raw1;
 	return;
 	}
 
-void cpkey(into)
-register unsigned long *into;
+void cpkey(unsigned long *into)
 {
 	register unsigned long *from, *endp;
 
@@ -125,8 +121,7 @@ register unsigned long *into;
 	return;
 	}
 
-void usekey(from)
-register unsigned long *from;
+void usekey(unsigned long *from)
 {
 	register unsigned long *to, *endp;
 
@@ -135,8 +130,7 @@ register unsigned long *from;
 	return;
 	}
 
-void des(inblock, outblock)
-unsigned char *inblock, *outblock;
+void des(unsigned char *inblock, unsigned char *outblock)
 {
 	unsigned long work[2];
 
@@ -146,9 +140,7 @@ unsigned char *inblock, *outblock;
 	return;
 	}
 
-static void scrunch(outof, into)
-register unsigned char *outof;
-register unsigned long *into;
+static void scrunch(unsigned char *outof, unsigned long *into)
 {
 	*into	 = (*outof++ & 0xffL) << 24;
 	*into	|= (*outof++ & 0xffL) << 16;
@@ -161,9 +153,7 @@ register unsigned long *into;
 	return;
 	}
 
-static void unscrun(outof, into)
-register unsigned long *outof;
-register unsigned char *into;
+static void unscrun(unsigned long *outof, unsigned char *into)
 {
 	*into++ = (*outof >> 24) & 0xffL;
 	*into++ = (*outof >> 16) & 0xffL;
@@ -320,8 +310,7 @@ static unsigned long SP8[64] = {
 	0x10041040L, 0x00041000L, 0x00041000L, 0x00001040L,
 	0x00001040L, 0x00040040L, 0x10000000L, 0x10041000L };
 
-static void desfunc(block, keys)
-register unsigned long *block, *keys;
+static void desfunc(unsigned long *block, unsigned long *keys)
 {
 	register unsigned long fval, work, right, leftt;
 	register int round;
diff --git a/region.c b/region.c
index 54603b7..9445755 100644
--- a/region.c
+++ b/region.c
@@ -223,8 +223,7 @@ RegionRec   miBrokenRegion = { { 0, 0, 0, 0 }, &miBrokenData };
 
 #ifdef DEBUG
 int
-miPrintRegion(rgn)
-    RegionPtr rgn;
+miPrintRegion(RegionPtr rgn)
 {
     int num, size;
     register int i;
@@ -244,9 +243,7 @@ miPrintRegion(rgn)
 }
 
 Bool
-miRegionsEqual(reg1, reg2)
-    RegionPtr reg1;
-    RegionPtr reg2;
+miRegionsEqual(RegionPtr reg1, RegionPtr reg2)
 {
     int i;
     BoxPtr rects1, rects2;
@@ -269,8 +266,7 @@ miRegionsEqual(reg1, reg2)
 }
 
 Bool
-miValidRegion(reg)
-    RegionPtr reg;
+miValidRegion(RegionPtr reg)
 {
     register int i, numRects;
 
@@ -324,9 +320,7 @@ miValidRegion(reg)
  *****************************************************************/
 
 RegionPtr
-miRegionCreate(rect, size)
-    BoxPtr rect;
-    int size;
+miRegionCreate(BoxPtr rect, int size)
 {
     register RegionPtr pReg;
    
@@ -358,10 +352,7 @@ miRegionCreate(rect, size)
  *****************************************************************/
 
 void
-miRegionInit(pReg, rect, size)
-    RegionPtr pReg;
-    BoxPtr rect;
-    int size;
+miRegionInit(RegionPtr pReg, BoxPtr rect, int size)
 {
     if (rect)
     {
@@ -382,8 +373,7 @@ miRegionInit(pReg, rect, size)
 }
 
 void
-miRegionDestroy(pReg)
-    RegionPtr pReg;
+miRegionDestroy(RegionPtr pReg)
 {
     good(pReg);
     xfreeData(pReg);
@@ -392,16 +382,14 @@ miRegionDestroy(pReg)
 }
 
 void
-miRegionUninit(pReg)
-    RegionPtr pReg;
+miRegionUninit(RegionPtr pReg)
 {
     good(pReg);
     xfreeData(pReg);
 }
 
 Bool
-miRegionBreak (pReg)
-    RegionPtr pReg;
+miRegionBreak (RegionPtr pReg)
 {
     xfreeData (pReg);
     pReg->extents = miEmptyBox;
@@ -451,9 +439,7 @@ miRectAlloc(
 }
 
 Bool
-miRegionCopy(dst, src)
-    register RegionPtr dst;
-    register RegionPtr src;
+miRegionCopy(RegionPtr dst, RegionPtr src)
 {
     good(dst);
     good(src);
@@ -911,8 +897,7 @@ miRegionOp(
  *-----------------------------------------------------------------------
  */
 void
-miSetExtents (pReg)
-    register RegionPtr pReg;
+miSetExtents (register RegionPtr pReg)
 {
     register BoxPtr pBox, pBoxEnd;
 
@@ -1018,10 +1003,10 @@ miIntersectO (
 
 
 Bool
-miIntersect(newReg, reg1, reg2)
-    register RegionPtr 	newReg;     /* destination Region */
-    register RegionPtr 	reg1;
-    register RegionPtr	reg2;       /* source regions     */
+miIntersect(
+    register RegionPtr 	newReg,     /* destination Region */
+    register RegionPtr 	reg1,
+    register RegionPtr	reg2)       /* source regions     */
 {
     good(reg1);
     good(reg2);
@@ -1173,10 +1158,10 @@ miUnionO (
 }
 
 Bool 
-miUnion(newReg, reg1, reg2)
-    RegionPtr		newReg;                  /* destination Region */
-    register RegionPtr 	reg1;
-    register RegionPtr	reg2;             /* source regions     */
+miUnion(
+    RegionPtr		newReg,                  /* destination Region */
+    register RegionPtr 	reg1,
+    register RegionPtr	reg2)             /* source regions     */
 {
     Bool overlap; /* result ignored */
 
@@ -1272,9 +1257,7 @@ miUnion(newReg, reg1, reg2)
  *
  */
 Bool
-miRegionAppend(dstrgn, rgn)
-    register RegionPtr dstrgn;
-    register RegionPtr rgn;
+miRegionAppend(RegionPtr dstrgn, RegionPtr rgn)
 {
     int numRects, dnumRects, size;
     BoxPtr new, old;
@@ -1458,9 +1441,9 @@ QuickSortRects(
  */
 
 Bool
-miRegionValidate(badreg, pOverlap)
-    RegionPtr badreg;
-    Bool *pOverlap;
+miRegionValidate(
+    RegionPtr badreg,
+    Bool *pOverlap)
 {
     /* Descriptor for regions under construction  in Step 2. */
     typedef struct {
@@ -1649,10 +1632,10 @@ bail:
 }
 
 RegionPtr
-miRectsToRegion(nrects, prect, ctype)
-    int			nrects;
-    register xRectangle	*prect;
-    int			ctype;
+miRectsToRegion(
+    int			nrects,
+    register xRectangle	*prect,
+    int			ctype)
 {
     register RegionPtr	pRgn;
     register RegDataPtr	pData;
@@ -1874,10 +1857,10 @@ miSubtractO (
  *-----------------------------------------------------------------------
  */
 Bool
-miSubtract(regD, regM, regS)
-    register RegionPtr	regD;               
-    register RegionPtr 	regM;
-    register RegionPtr	regS;          
+miSubtract(
+    register RegionPtr	regD,
+    register RegionPtr 	regM,
+    register RegionPtr	regS)          
 {
     Bool overlap; /* result ignored */
 
@@ -1939,10 +1922,10 @@ miSubtract(regD, regM, regS)
  *-----------------------------------------------------------------------
  */
 Bool
-miInverse(newReg, reg1, invRect)
-    RegionPtr 	  newReg;       /* Destination region */
-    RegionPtr 	  reg1;         /* Region to invert */
-    BoxPtr     	  invRect; 	/* Bounding box for inversion */
+miInverse(
+    RegionPtr 	  newReg,       /* Destination region */
+    RegionPtr 	  reg1,         /* Region to invert */
+    BoxPtr     	  invRect) 	/* Bounding box for inversion */
 {
     RegionRec	  invReg;   	/* Quick and dirty region made from the
 				 * bounding box */
@@ -1999,9 +1982,9 @@ miInverse(newReg, reg1, invRect)
  */
 
 int
-miRectIn(region, prect)
-    register RegionPtr  region;
-    register BoxPtr     prect;
+miRectIn(
+    register RegionPtr  region,
+    register BoxPtr     prect)
 {
     register int	x;
     register int	y;
@@ -2095,10 +2078,10 @@ miRectIn(region, prect)
 */
 
 void
-miTranslateRegion(pReg, x, y)
-    register RegionPtr pReg;
-    register int x;
-    register int y;
+miTranslateRegion(
+    register RegionPtr pReg,
+    register int x,
+    register int y)
 {
     int x1, x2, y1, y2;
     register int nbox;
@@ -2209,9 +2192,9 @@ miRegionDataCopy(
 }
 
 void
-miRegionReset(pReg, pBox)
-    RegionPtr pReg;
-    BoxPtr pBox;
+miRegionReset(
+    RegionPtr pReg,
+    BoxPtr pBox)
 {
     good(pReg);
     assert(pBox->x1<=pBox->x2);
@@ -2222,10 +2205,11 @@ miRegionReset(pReg, pBox)
 }
 
 Bool
-miPointInRegion(pReg, x, y, box)
-    register RegionPtr pReg;
-    register int x, y;
-    BoxPtr box;     /* "return" value */
+miPointInRegion(
+    register RegionPtr pReg,
+    register int x,
+    register int y,
+    BoxPtr box)     /* "return" value */
 {
     register BoxPtr pbox, pboxEnd;
     int numRects;
@@ -2256,8 +2240,7 @@ miPointInRegion(pReg, x, y, box)
 }
 
 Bool
-miRegionNotEmpty(pReg)
-    RegionPtr pReg;
+miRegionNotEmpty(RegionPtr pReg)
 {
     good(pReg);
     return(!REGION_NIL(pReg));
@@ -2271,8 +2254,7 @@ miRegionBroken(RegionPtr pReg)
 }
 
 void
-miRegionEmpty(pReg)
-    RegionPtr pReg;
+miRegionEmpty(RegionPtr pReg)
 {
     good(pReg);
     xfreeData(pReg);
@@ -2282,8 +2264,7 @@ miRegionEmpty(pReg)
 }
 
 BoxPtr
-miRegionExtents(pReg)
-    RegionPtr pReg;
+miRegionExtents(RegionPtr pReg)
 {
     good(pReg);
     return(&pReg->extents);
@@ -2525,8 +2506,7 @@ miClipSpans(
 
 /* find the band in a region with the most rectangles */
 int
-miFindMaxBand(prgn)
-    RegionPtr prgn;
+miFindMaxBand(RegionPtr prgn)
 {
     register int nbox;
     register BoxPtr pbox;
-- 
2.48.1

